src/Backend/Haskell.idr

```diff
@@ -1,68 +1,107 @@
 module Backend.Haskell
 
-import Data.Vect
 import Control.Monad.State
+import Text.PrettyPrint.WL
 
 import Backend.Utils
 import Types
 import Typedefs
 
-%default partial
+import Data.Vect
+
+%default total
 %access public export
 
-guardPar : String -> String
-guardPar str = if any isSpace $ unpack str then parens str else str
+data HType : Type where
```

Comment:
I think `HType` could do with a doc comment. And perhaps a different name? From seeing this now for the 1st time I'm assuming this means 'Haskell type' (and not e.g. heterogeneous type) so perhaps `HaskellType` is too long (?), and single letters are admittedly noice, but how do we feel about eg `HsType`? Just throwing this out there...